knowledge-base/create-custom-image-properties-resolver-net-standard.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Create Custom ImagePropertiesResolver in .Net Standard
+description: Describes how to create a custom implementation of ImagePropertiesResolver in .Net Standard.
+type: how-to
+page_title: Create custom implementation of ImagePropertiesResolverBase in .Net Standard
+slug: create-custom-image-properties-resolver-net-standard
+position: 0
+tags: image, resolver, jpeg, imagepropertiesresolver, netstandard
+res_type: kb
+---
+
+<table>
+<thead>
+	<tr>
+		<th>Product Version</th>
+		<th>Product</th>
+		<th>Author</th>
+	</tr>
+</thead>
+<tbody>
+	<tr>
+		<td>2023.1.315</td>
+		<td>RadPdfProcessing</td>
+		<td><a href="https://www.telerik.com/blogs/author/yoan-karamanov">Yoan Karamanov</a></td>
+	</tr>
+</tbody>
+</table>
+
+## Description
+
+**.NET Standard** specification does not define APIs for converting images or scaling their quality. That is why to export to PDF format a document containing images different than Jpeg and Jpeg2000 or ImageQuality different than High, you will need to provide an implementation of the **ImagePropertiesResolver** abstract class. This property enables you to set a resolver implementation that can parse the image raw data to separate its colors and alpha channel. This implementation should be passed to the **ImagePropertiesResolver** property of the of **FixedExtensibilityManager**.
+
+## Solution
+
+The following code snippets demonstrates how to create a custom implementation of the ImagePropertiesResolver abstract class using the [SixLabors.ImageSharp](https://github.com/SixLabors/ImageSharp) library and set it to the ImagePropertiesResolver property of the FixedExtensibilityManager.
+
+#### __[C#] Create a custom implementation inheriting the ImagePropertiesResolverBase abstract class__
+
+{{region kb-create-custom-image-properties-resolver1}}
+public class ImagePropertiesResolver : ImagePropertiesResolverBase
+{
+    public override TelerikSize GetImageSize(byte[] imageData)
+    {
+        TelerikSize size = TelerikSize.Empty;
+        try
+        {
+            using (ImageSharp image = ImageSharp.Load(imageData))
+            {
+                size = new TelerikSize(image.Width, image.Height);
```

Comment:
You can change it to the whole namespace because when you don't provide usings the clients could have difficulties finding out what TelerikSize is.